File: tests/fixtures/resend/resend-batch-size-not-enforced-broken/01-direct-request-array.ts
import { Resend } from 'resend';
import { NextResponse } from 'next/server';

const resend = new Resend(process.env.RESEND_API_KEY);

// Sends the request-provided array straight to batch.send with no size guard.
export async function POST(req: Request) {
  const { emails } = await req.json();

  const { data, error } = await resend.batch.send(emails);

  if (error) {
    return NextResponse.json({ error: error.message }, { status: 500 });
  }

  return NextResponse.json({ ids: data?.data?.map((r) => r.id) });
}

File: tests/fixtures/resend/resend-batch-size-not-enforced-broken/01-direct-request-array.py
import os
import resend
from flask import Flask, jsonify, request

app = Flask(__name__)
resend.api_key = os.environ["RESEND_API_KEY"]


@app.route("/batch", methods=["POST"])
def send_batch():
    emails = request.get_json()["emails"]
    # Variable array with no len() guard — should flag.
    result = resend.Batch.send(emails)
    return jsonify(result)
